sample_workloads/nccltest/README.md

```diff
@@ -0,0 +1,151 @@
+# TCPX NCCL Level Benchmarks
+
+This document will walk you through how to build, push, and deploy the nccl-benchmark image for use on a GKE cluster configured with an A3 nodepool.
+
+## Building the Benchmark Docker Image
+
+To build the latest benchmark docker image, run:
+
+```shell
+cd docker && docker build . -t nccl-benchmarks
+```
+
+**Note: A pre-built image is available in the values.yaml file**
+
+## Running the TCPX NCCL Benchmarks
+
+This section describes how you can run a 2-node world-level all-gather
+benchmark at message sizes 1G and 8G.
+
+
+If you intend to run with GKE, run:
+
+```shell
+cd gke
+PARAMS="cluster.nNodes=2,"
+PARAMS+="ncclBenchmarks.benchmarks=AllGather,"
+PARAMS+="ncclBenchmarks.masks=0x0,"
+PARAMS+="ncclBenchmarks.msgSizes=1G\,8G"
+helm install "${USER}-nccl-bm" . --set "$PARAMS"
+```
+
+Once the job is scheduled, find your master pod by running
+
+```shell
+kubectl get pods | grep "${USER}-nccl-bm.*pod0"
+```
+
+You can then follow the logs with
+
+```shell
+kubectl logs --follow <master-pod-name> -c nccl-benchmarks
+```
+
+### Finding Results
+
+The container will log the output of Nvidia's nccl-test binaries for each of the tests that are requested in the parameters.
+
+Each test run is logged separately. If you specify a release that uses multiple tests and multiple runs, the logs will be ordered by Test0Run0...Test0RunN...TestNRunN.
+
+An example test output would look like:
+```
+benchmark: all_reduce_perf, mask: 0x0, run 1/1
+# nThread 1 nGpus 1 minBytes 1048576 maxBytes 8589934592 step: 2(factor) warmup iters: 2 iters: 2 agg iters: 1 validation: 0 graph: 0
```

Comment:
Maybe also attach a run with more iterations, to reduce variance? But this is only for demonstration purposes, so I don't think it's a big problem.